### PR TITLE
Arrays: translation of flatten, isList, normalize, pick, some, every, map

### PR DESCRIPTION
**Pull request by landsman**
the first pehapkari quest

### CODE REVIEW

**Comment by f3l1x** [comment]
Super, diky moc. 

1. Potrebuje to rebase na aktualni doc-2.4.
2. Mrkni na code-checker a odstran prebytecny white spaces.

:-)

File: en/arrays.texy [hunk 1]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+
+Transform multidimensional array to flat array.
+
+/--php
+$array = Arrays::flatten([
+	2 => ['a', ['b']],
+	4 => ['c', 'd'],
+	'e',
+]);
+// $array = [1 => 'a', 2 => 'b', 3 => 'c', 4 => 'd'];
```
Comment (dg) [inline en/arrays.texy]: `$array = ['a', 'b', 'c', 'd']`

File: en/arrays.texy [hunk 2]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+\--
+
+
+isList(array $array): bool .[method]
+------------------------------------
+
+Check if the array is indexed by numeric keys (as list).
```
Comment (dg) [inline en/arrays.texy]: …in descending series from zero…

File: en/arrays.texy [hunk 3]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+\--
+
+
+normalize(array $arr, string $filling = null): array .[method]
+--------------------------------------------------------------
+
+Replace numeric keys in array to the value from variable `$filling`.
+
+/--php
+$array = Arrays::normalize([1 => 'first', 'a' => 'second']);
```
Comment (dg) [inline en/arrays.texy]: better example is without numeric indexed, ie `Arrays::normalize(['first', 'a' => 'second']);` or `Arrays::normalize(['first', 'a' => 'second', 'third', `x` => `fourth`]);`

File: en/arrays.texy [hunk 4]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+/--php
+$array = Arrays::normalize([1 => 'first', 'a' => 'second'], 'foobar');
+// $array = ['first' => 'foobar', 'a' => 'second'];
+\--
+
+
+pick(array &$arr, string|int $key, mixed $default = null): mixed .[method]
+--------------------------------------------------------------------------
+
+Returns value of the item in array or if not exist throw exception or return filled `$default`.
```
Comment (dg) [inline en/arrays.texy]: Returns and removes

File: en/arrays.texy [hunk 5]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+// have only bar???
+$res = Arrays::some($array, function($value, $key, $arr) { return $v === 'bar'; });
+// $res = false
+\--
+
+
+map(array $arr, callable $callback): array .[method]
+----------------------------------------------------
+
+Apply `$callback` to the all items in array.
```
Comment (dg) [inline en/arrays.texy]: … and return results.

File: en/arrays.texy (hunk 2 above)
Comment (nufue) [inline en/arrays.texy]: ...in a continuous ascending sequence starting from zero...

File: en/arrays.texy (hunk 5 above)
Comment (nufue) [inline en/arrays.texy]: Applies `$callback` to each item of the `$arr` and returns resulting array.

File: en/arrays.texy [hunk 6]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+// have some bar???
+$res = Arrays::some($array, function($value, $key, $arr) { return $v === 'bar'; });
+// $res = true
+\--
+
+
+every(array $arr, callable $callback): bool .[method]
+-----------------------------------------------------
+
+Check if all items of the array go through condition in `$callback`.
```
Comment (nufue) [inline en/arrays.texy]: Checks whether all items of the `$arr` satisfy the condition in `$callback`.

File: en/arrays.texy [hunk 7]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+// $b = 'foobar'
+$c = Arrays::pick($array, 'not-exists');
+// throws Nette\InvalidArgumentException
+\--
+
+
+some(array $arr, callable $callback): bool .[method]
+----------------------------------------------------
+
+Check if some item of the array go through condition in `$callback`.
```
Comment (nufue) [inline en/arrays.texy]: Checks whether any item of the `$arr` satisfies a condition in `$callback`.

File: en/arrays.texy (hunk 3 above)
Comment (landsman) [inline en/arrays.texy]: I think that I don't understand what you mean :/ I was started from czech version: https://github.com/nette/docs/blob/doc-2.4/cs/arrays.texy#L160 ... what do you suggest?

File: en/arrays.texy (hunk 2 above)
Comment (landsman) [inline en/arrays.texy]: added https://github.com/nette/docs/pull/584/commits/8967354d6ad91396424b7785b443bbed7cf4917a#diff-8ed9a00fefc55fc0e8371c84d1020c49R151

File: en/arrays.texy (hunk 3 above)
Comment (dg) [inline en/arrays.texy]: I mean that `1 =>` in example is confusing.

File: en/arrays.texy [hunk 8]
```diff
@@ -129,4 +129,101 @@ Arrays::renameKey($array, 'first', 'renamed');
...
+
+isList(array $array): bool .[method]
+------------------------------------
+
+Check if the array is indexed by numeric keys in descending series from zero.
+
+/--php
+Arrays::isList(['a', 'b', 'c'])); // true
+Arrays::isList([4 => 1, 2, 3])); // false
+\--
```
Comment (dg) [inline en/arrays.texy]: Can you please add third example `Arrays::isList(['a' => 1, 'b' => 2])); // false`?

**Comment by dg** [comment]
Thank you!

**Comment by landsman** [comment]
Thank you for merge, I'll continue ;)

**Pull request merged**